verilog/analysis/verilog_linter_configuration.cc

```diff
@@ -83,6 +83,46 @@ std::string ProjectPolicy::ListPathGlobs() const {
                        });
 }
 
+bool RuleBundle::ParseConfiguration(absl::string_view text, std::string *error) {
+  // Clear the vector to overwrite any existing value.
+  rules.clear();
+
+  for (absl::string_view part : absl::StrSplit(text, absl::ByAnyChar(",\n"),
+                                               absl::SkipEmpty())) {
+    // If prefix is '-', the rule is disabled.
+    // Note that part is guaranteed to be at least one character because
+    // of absl::SkipEmpty()
+    const bool prefix_minus = (part[0] == '-');
+
+    RuleSetting setting = {!prefix_minus, ""};
+
+    const auto rule_name_with_config = part.substr(prefix_minus ? 1 : 0);
+
+    // Independent of the enabled-ness: extract a configuration string
+    // if there is any assignment.
+    const auto equals_pos = rule_name_with_config.find('=');
+    if (equals_pos != absl::string_view::npos) {
+      const auto config = rule_name_with_config.substr(equals_pos + 1);
+      setting.configuration.assign(config.data(), config.size());
+    }
+    const auto rule_name = rule_name_with_config.substr(0, equals_pos);
+    const auto rule_name_set = analysis::GetAllRegisteredLintRuleNames();
+    const auto rule_iter = rule_name_set.find(rule_name);
+
+    // Check if text is a valid lint rule.
+    if (rule_iter == rule_name_set.end()) {
+      *error = absl::StrCat("invalid flag \"", rule_name, "\"");
+      return false;
+    } else {
+      // Map keys must use canonical registered string_views for guaranteed
+      // lifetime, not just any string-equivalent copy.
+      rules[*rule_iter] = setting;
+    }
+  }
+
+  return true;
+}
+
```

Comment:
Note that I didn't move the tests to apply to this yet. Should the tests be just copied?

Reply:
@fangism ?

Reply:
I would move-adapt the tests to avoid redundancy.

Reply:
Sure, moved, thanks!